### ISSUE TITLE
Ability to exclude only certain lines

### ISSUE THREAD

**Issue by woarewe** [rubocop-hq/rubocop#7683]
Hi everyone!

Could you please suggest to me about the future features of the project. Rubocop is a great tool when you start a new project, you configure your cops and your team follows them. It's amazing, but if you wanna introduce it in legacy project and run it with the param `--auto-gen-config` it creates `.rubocop.todo.yml` that disables copes for a whole file. It works great when your new code is in new files, but if you change already excluded file rubocop doesn't check it.

Maybe, are you planning to add the ability to exclude only certain lines in the way like this?
```yml
  Layout/MultilineMethodCallBraceLayout:
    Exclude:
      - 'app/controllers/users.rb:34'
```

Or It will be much better and easier to add ability to paste inline comments automatically.

If you need any type of help when you will develop this feature I will be glad to help.
Thanks for your time and attention.

**Comment by Lykos** [comment]
Using line numbers in long-living configs is a bad idea. They will inevitably change and keeping them updated would be a burden to whoever changes those file. Line numbers are just too brittle. One little deletion above the line in question breaks it.

I think excluding certain parts of a file should use the existing enable/disable mechanism.

Note that I am not a maintainer and I have no authority on this. That's just my opinion when looking at this.

**Comment by woarewe** [comment]
I agree with you about the config, that's hard to maintain.
But I see at least two solutions:
1) Add a run mode that can paste inline comments. It's not the best practice just like # TODO comments but if we use it in the right way it will give to us a lot of benefits.
2) When it adds excluding cops to `.rubocop.todo.yml` the file also needs to add some hash for checking, if this line already changed and the current hash sum is not equal to the hash sum in exclude file just ignore this exclude cop.

Rubocop, just like other tools, helps us to solve our problems. A big part of the ruby community considers that a lot of ruby functionality is redundant, for example, like refinements, if you don't like it, just don't use it. If my proposal is accepted and one time is released and it helps to solve someone's problems I will be glad.  But if my offer is declined, it will be ok-ish, because I just described my problem and maybe someone also has the same.

**Comment by Lykos** [comment]
I agree that something like that is useful, I just don't think that the initial suggestion is a good idea. But if you polish it along the lines of what you suggested, it might actually become useful.

**Comment by brandonjmckay** [comment]
I'd love to see something like what you mention there @woarewe ! Either the hash checking, or inline comments as possible ways of generating a todo would be amazing.
